### DISCUSSION TITLE
Date and time data type time for filtering.

### DISCUSSION THREAD

**Discussion by wammy19** [qdrant/qdrant#43] [Q&A] [answered]
Hi,

I'm just wondering if there is a way of filtering based on date and time range?

**Comment by generall** [comment]
Hi, right now you can store date in a timestamp format as integer and use range filter for that. Support for more convenient date handling will definitely be added later. 

If you have any specific feature request you are welcome to describe it here or in the issues section

**Comment by JeyDi** [comment]
Hi @generall Is there any news about support for date and time in the payload? What about having a dedicated index?
Thank you!

**Reply by timvisee** [reply]
Would using a timestamp be sufficient, as suggested above? You could simply use the `int` index for that.

**Reply by timvisee** [reply]
We do support a date time index since Qdrant 1.8.

Documentation: https://qdrant.tech/documentation/concepts/payload/#datetime

**Reply by luisiacc** [reply]
> no that isn't sufficient for a lot of use cases where you want an LLM to be able to figure out the date for "last week" or last month. It is able to do that based off of a date like 06-04-2024 but not a timestamp. What would be the best way to handle that? I think this hasn't been resolved for 3 years now?

you could convert the timestamp to a date beforehand right?

**Comment by timvisee** [comment, marked as answer]
Qdrant 1.8 supports a date time index: https://qdrant.tech/documentation/concepts/payload/#datetime

Alternatievely you can use a timestamp with an integer index as described here: https://github.com/orgs/qdrant/discussions/43#discussioncomment-911581
